theano/tensor/nlinalg.py

```diff
@@ -742,14 +742,18 @@ def __init__(self, ind=2):
     def make_node(self, a):
         a = as_tensor_variable(a)
         out_dtype = a.dtype
-        out = theano.tensor.matrix(dtype=out_dtype)
+        out = theano.tensor.tensor4(dtype=out_dtype)
```

Comment:
out isn't always a 4d tensor. For example:

python -c "import numpy as np; print np.linalg.tensorinv(np.random.rand(2,15,30)).shape"
(30, 2, 15)

python -c "import numpy as np; print np.linalg.tensorinv(np.random.rand(30,2,15),ind=1).shape"
(2, 15, 30)

I think out should be: `out=a.type()`

Add those 2 cases as a test.